app/src/main/java/com/example/househomey/HomeFragment.java

```diff
@@ -236,4 +286,59 @@ private void updateListData() {
         this.listSumView.setText("$" + listSum.toString());
         this.listCountView.setText(Integer.toString(listCount));
     }
+
+    /**
+     * Displays the sort menu with properties of items to sort by
+     * @param view The view on which the pop up menu is displayed
+     * @param sortProperties A mapping of properties to their comparators
+     */
+    private void showSortMenu(View view, Map<String, Comparator<Item>> sortProperties) {
+        PopupMenu popupMenu = new PopupMenu(requireContext(), view);
+        MenuInflater inflater = popupMenu.getMenuInflater();
+        inflater.inflate(R.menu.sort, popupMenu.getMenu());
+
+        sortOrder = toggleOrder.isChecked();
+        popupMenu.setOnMenuItemClickListener(item -> {
+            int itemId = item.getItemId();
+            if (itemId == R.id.sort_by_description) {
+                currentSort = sortProperties.get("description");
+                sortItems(filteredItemList, currentSort, sortOrder);
+            } else if (itemId == R.id.sort_by_date) {
+                currentSort = sortProperties.get("date");
+                sortItems(filteredItemList, currentSort, sortOrder);
+            } else if (itemId == R.id.sort_by_make) {
+                currentSort = sortProperties.get("make");
+                sortItems(filteredItemList, currentSort, sortOrder);
+            } else if (itemId == R.id.sort_by_estimatedValue) {
+                currentSort = sortProperties.get("cost");
+                sortItems(filteredItemList, currentSort, sortOrder);
+            } else {
+                return false;
+            }
+            return true;
+        });
+
+        popupMenu.show();
+    }
+
+    /**
+     * Sorts the item list on the homepage based the provided Comparator in either descending
+     * or ascending order
+     * @param itemList List of items to sort
+     * @param currentSort Comparator for the property by which we are sorting
+     * @param descending Whether to sort in descending order or not
+     */
+    private void sortItems(ArrayList<Item> itemList, Comparator<Item> currentSort, boolean descending) {
+
+        if (!descending){
+            itemList.sort(currentSort);
+        }
+        else {
+            itemList.sort(Collections.reverseOrder(currentSort));
+        }
+        itemAdapter.notifyDataSetChanged();
+    }
```

Comment:
Any reason why we are using arguments here? These are all class properties right?

Reply:
Oh yeah, my bad. They weren't properties when I first started implementing this method. Was too lazy to switch it back. Will change it to have no arguments